HW2_2

Balan666 opened the pull request:
Added code, ipynb (the third file with corruption was analysed there), the results of analysing another two files in a separate directory

Balan666 committed:
Added code for HW2_2 and output files

Code as .py and .ipynb
In ipynb the error management can be seen

Balan666 committed:
Rename DNA_stats.txt to HW2_2/output/DNA_stats.txt

moved it to a directory

Balan666 committed:
Rename Protein_stats.txt to HW2_2/output/Protein_stats.txt

moved file to a directory

Balan666 committed:
Rename ITMO_PythonHW2_2_Balan.ipynb to HW2_2/ITMO_PythonHW2_2_Balan.ipynb

Moved file to a directory

Balan666 committed:
Rename itmo_pythonhw2_2_balan.py to HW2_2/itmo_pythonhw2_2_balan.py

Moved file to the directory

Balan666 commented:
@rinnifox

rinnifox commented:
Hi Anna! Very nice solution with writing text files with the analysis.

I have a few comments though:
1. For the third file I see `list index out of range` error that comes from `parse_seqkit_output()`, but in fact it would be much beneficial to either just report that the seqkit output is empty (meaning some error occurred) or to capture this exact error using `stderr`.  In case of catching an error in stderr(=getting empty output), you can just finish your program with something like `Seqkit was not able to process your fasta file` so you don't need to proceed further :)
2. It is recommended to be very cautious when using globals. And moreover, I don't really get the idea behind using global in your case. You are using it only once without passing to another function. So I guess you can just remove it?
3. Output file naming. I'd suggest you name the txt files after the input file like `hw_file1_stats.txt`. Otherwise if I'd like to compute some files other than from HW, the output will be overwritten with DNA/Protein stats.txt . And for corrupted files (like the third one) we have an error text in the filename, which also looks weird :)

Balan666 committed:
fixed mistakes

1. Used sys.exit to stop running the program in case of an error
2. deleted global
3. changed the naming of files to hw_file1.fasta_stats.txt, hw_file2.fasta_stats.txt, etc

Balan666 commented:
@rinnifox 
I've corrected the mistakes (added the files with new code and output)

rinnifox commented:
Ok, nice!

Just one more comment: I still see `SystemExit: Something's wrong with your fasta file!` for hw_file3.fasta, which comes from `parse_response_uniprot()`. If something fails at the stage of running seqkit or parsing its result, there is no need to proceed further :) this is important from the optimization perspective.

So at this point:
```
    except Exception as e:
        sys.stderr.write("Seqkit was not able to process your fasta file\n")
        return None, str(e)
```
You can just terminate your script running.

And here:
```
    except subprocess.CalledProcessError as e:
        sys.stderr.write("Seqkit was not able to process your fasta file\n")
        sys.exit('Seqkit was not able to process your fasta file\n')
```

I think you can change the error description to smth like 'File is not found', as if I'm not mistaken this subprocess error appears right in this case (if the file do exist but smth wrong with it like in hw_file3.fasta, this error is not raised).

rinnifox commented:
This PR can be merged.
